src/main.ts

```diff
@@ -0,0 +1,62 @@
+import { abciQuery } from './abci-query';
+
+const fetchRPCAddr = async (netconfigURL: string) => {
+  const response = await fetch(netconfigURL, {
+    headers: { accept: 'application/json' },
+  });
+  const { rpcAddrs } = await response.json();
+
+  return rpcAddrs[Math.floor(Math.random() * rpcAddrs.length)];
+};
+
+const networkConfigUrl = (() => {
+  const agoricNetName =
+    new URLSearchParams(window.location.search).get('network') ?? 'main';
+
+  if (agoricNetName === 'local') {
+    return 'https://wallet.agoric.app/wallet/network-config';
+  } else {
+    return `https://${agoricNetName}.agoric.net/network-config`;
+  }
+})();
+
+const setMessage = (message: string) => {
+  document?.getElementById('msg')?.replaceChildren(message);
+  console.info(message);
+};
+
+const tryRedirect = async () => {
+  const rpc = await fetchRPCAddr(networkConfigUrl);
+  let endorsedUI;
+
+  try {
+    const data = await abciQuery(
+      rpc,
+      '/custom/vstorage/data/published.vaultFactory.governance'
+    );
+    const { values } = JSON.parse(data.value);
+    const value = JSON.parse(values[0]);
```

Comment:
is the first element of the array the most recent data?

Reply:
Seems like it's always an array with one element, for past data I'd assume you need to change blockheight in the query

Reply:
multiple data will come in the same block if there were multiple publishes. Unlikely in a governance parameter but before merging let's confirm which end of the array has the latest.

Reply:
Ahh I see, sounds good

Reply:
Fixed thanks for confirming